### DISCUSSION TITLE
Does Fallback in X-ray work only for TLS protocols? How about REALITY?

### DISCUSSION THREAD

**Discussion by houmie** [XTLS/Xray-core#2200] [Q&A]
I was reading the [documentation](https://xtls.github.io/en/document/level-1/fallbacks-with-sni.html#xray-configuration) about SNI fallbacks in XRay.

Do I assume correctly that fallbacks can only work if I use protocols based on TLS? Such as 
VLESS+gRPC+TLS, VLESS+WS+TLS or Trojan+gRPC+TLS or VLESS-TLS-XTSL-VISION etc, because of this settings?

```
"streamSettings": {
    "network": "tcp",
    "security": "tls",
```

1) So I couldn’t use non TLS based protocols such as VLESS-XTLS-VISION-REALITY?
2) Is it possible using fallbacks for REALITY based protocols at all? 
3) Can REALITY based fallbacks eventually fallback to a website as a disguise?  (I know it's possible with TLS based protocols, but I haven't found any example with REALITY) 

Many Thanks

**Comment by FranzKafkaYu** [comment]
> Do I assume correctly that fallbacks can only work if I use protocols based on TLS?

strictly it only supports in `tcp`+`tls`(including xtls-rprx-vision),plz refer to [this](https://xtls.github.io/config/features/fallback.html#fallbacks-%E9%85%8D%E7%BD%AE)    

unfortunately,If you use REALITY,you can't use fallback feature.

**Reply by houmie** [reply]
Thank you. That was very helpful.

**Comment by RPRX** [comment]
`fallbacks` 有两个功能，防主动探测和灵活分流，但是 REALITY 本身就防主动探测，所以它的 `fallbacks` 仅用于分流给其它内层协议

虽然 `fallbacks` 的 `path` 暂不支持 h2c，但你可以写多个 REALITY `serverNames`，Vision 按 `name` 分流到 H2 和 gRPC，三协议共存

**Reply by houmie** [reply]
Thank you for the information.

**Reply by evkravchenko** [reply]
> `fallbacks` 有两个功能，防主动探测和灵活分流，但是 REALITY 本身就防主动探测，所以它的 `fallbacks` 仅用于分流给其它内层协议
> 
> 虽然 `fallbacks` 的 `path` 暂不支持 h2c，但你可以写多个 REALITY `serverNames`，Vision 按 `name` 分流到 H2 和 gRPC，三协议共存

Hello
How about using inbound with reality as fallback for the other inbound with reality? I my case i want to use seamless reality keys rotation when key changes in the front inbound and the old key from him move to fallback inbound. In theory, reality backup shouldn't work, but I decided to check... it really doesn't work.
Is it possible to using inbound with reality as a fallback inbound?

**Reply by Saeid008** [reply]
> `fallbacks` 有两个功能，防主动探测和灵活分流，但是 REALITY 本身就防主动探测，所以它的 `fallbacks` 仅用于分流给其它内层协议
> 
> 虽然 `fallbacks` 的 `path` 暂不支持 h2c，但你可以写多个 REALITY `serverNames`，Vision 按 `name` 分流到 H2 和 gRPC，三协议共存

Does fallback support Quic/HTTP3 ?
I think it's only support h2. since QUIC/H3 is based on UDP.
@Fangliding @yuhan6665 @RPRX

**Comment by hellofr13nd** [comment]
Thank you for your answers, so which version is better for now as an anti-active detection? Reality or TLS+Fallbacks?

**Comment by egasvegas1109** [comment]
Is there no way to use fallback for Reality now? x ray has taken over port 443, but I need to make a website for cover, I don't want to do it on port 80
